packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,843 @@
+          shift 1
+          ;;
+        verbose|insecure|noproxy|noreload|daemon-not-running)
+          NETDATA_CLAIM_EXTRA="${NETDATA_CLAIM_EXTRA} -${optname}"
+          ;;
+        *)
+          warning "Ignoring unrecognized claiming option ${optname}"
+          ;;
+      esac
+      ;;
+  esac
+  shift 1
+done
+
+check_claim_opts
+confirm_root_support
+get_system_info
+
+tmpdir="$(create_tmp_directory)"
+progress "Using ${tmpdir} as a temporary directory."
+cd "${tmpdir}" || exit 1
+
+handle_existing_install
+
+case "${SYSTYPE}" in
+  Linux)
+    if [ "${NETDATA_ONLY_STATIC}" -ne 1 ]; then
+      try_package_install
+
+      case "$?" in
+        0)
+          NETDATA_INSTALL_SUCCESSFUL=1
+          ;;
+        1)
+          fatal "Unable to install on this system."
+          ;;
+        2)
+          if [ "${NETDATA_ONLY_NATIVE}" -eq 1 ]; then
+            fatal "Could not install native binary packages."
+          else
+            warning "Could not install native binary packages, falling back to alternative installation method."
+          fi
+          ;;
+      esac
+    fi
+
+    if [ "${NETDATA_ONLY_NATIVE}" -ne 1 ] && [ -z "${NETDATA_INSTALL_SUCCESSFUL}" ]; then
+      try_static_install
+
+      case "$?" in
+        0)
+          NETDATA_INSTALL_SUCCESSFUL=1
+          NETDATA_USER_CONFIG_DIR="/opt/netdata/etc/netdata"
+          ;;
+        1)
+          fatal "Unable to install on this system."
+          ;;
+        2)
+          if [ "${NETDATA_ONLY_STATIC}" -eq 1 ]; then
+            fatal "Could not install static build."
+          else
+            warning "Could not install static build, falling back to alternative installation method."
+          fi
+          ;;
+      esac
+    fi
+    ;;
+  Darwin)
+    if [ "${NETDATA_ONLY_NATIVE}" -eq 1 ]; then
+      fatal "User requested native package, but native packages are not available for macOS. Try installing without \`--only-native\` option."
+    elif [ "${NETDATA_ONLY_STATIC}" -eq 1 ]; then
+      fatal "User requested static build, but static builds are not available for macOS. Try installing without \`--only-static\` option."
+    else
+      fatal "This script currently does not support installation on macOS."
+    fi
+    ;;
+  FreeBSD)
+    if [ "${NETDATA_ONLY_NATIVE}" -eq 1 ]; then
+      fatal "User requested native package, but native packages are not available for FreeBSD. Try installing without \`--only-native\` option."
+    elif [ "${NETDATA_ONLY_STATIC}" -eq 1 ]; then
+      fatal "User requested static build, but static builds are not available for FreeBSD. Try installing without \`--only-static\` option."
+    else
+      fatal "This script currently does not support installation on FreeBSD."
+    fi
+    ;;
+esac
+
+if [ -n "${NETDATA_DISABLE_TELEMETRY+x}" ]; then
+  run touch "${NETDATA_USER_CONFIG_DIR}/.opt-out-from-anonymous-statistics"
+fi
+
+if [ -n "${NETDATA_CLAIM_TOKEN}" ]; then
+  claim
```

Comment:
I think there are code paths that end up calling this function multiple times. Eg. when hitting the `claim` line 394.

Reply:
Nice catch, I forgot to have it explicitly exit in some of those cases.

Reply:
The flip side of that is that it makes handling of existing installs more complex, because then we also need a special flag to _skip_ the normal install code and can’t rely on early exits.